lib/evaluateFunctions.js

```diff
@@ -291,53 +289,50 @@ function groupTermsByRoot(node) {
     i = j;
   }
 
-  // if we don't group any factors, then we can't simplify it any more
   if (hasGroups) {
     newNode.args[0] = children.length === 1 ?
         children[0] : NodeCreator.operator('*', children);
     return NodeStatus.nodeChanged(
       MathChangeTypes.GROUP_TERMS_BY_ROOT, node, newNode);
   }
+  // if we don't group any factors, then we can't simplify it any more
   return NodeStatus.noChange(node);
 }
 
-// Given an nthRoot node with a constant positive root, will group the arguments
+// Given an nthRoot node with a constant positive root,
 // will convert any grouped factors into exponent nodes as a step
 // e.g. nthRoot((2 * 2) * 2, 2) -> nthRoot(2^2 * 2, 2)
```

Comment:
can you add a disclaimer that groups are expected to be the same length as the root number?